hpms/src/main/java/gov/cms/ab2d/hpms/service/AttestationUpdaterServiceImpl.java

```diff
@@ -0,0 +1,152 @@
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Primary
+@Service
+public class AttestationUpdaterServiceImpl implements AttestationUpdaterService {
+
+    private static final int BATCH_SIZE = 100;
+
+    private final HPMSFetcher hpmsFetcher;
+
+    private final SponsorRepository sponsorRepository;
+
+    private final ContractRepository contractRepository;
+
+//    private static final String HOURLY = "0 0 0/1 1/1 * ?";
+
+    @Autowired
+    public AttestationUpdaterServiceImpl(SponsorRepository sponsorRepository, ContractRepository contractRepository,
+                                         HPMSFetcher hpmsFetcher) {
+        this.sponsorRepository = sponsorRepository;
+        this.contractRepository = contractRepository;
+        this.hpmsFetcher = hpmsFetcher;
+    }
+
+/*  todo: remove the comments and enable this code when integrating with the real hpms service
+    @Scheduled(cron = AttestationUpdaterService.HOURLY)
+    public void pollHmsData() {
+        pollOrganizations();
+    }
+ */
+
+    @Override
+    public void pollOrganizations() {
+        hpmsFetcher.retrieveSponsorInfo(this::processOrgInfo);
+    }
+
+    private void processOrgInfo(HPMSOrganizations orgInfo) {
+        Map<String, Contract> existingMap = buildExistingContractMap();
+        // detect new Contracts
+        List<HPMSOrganizationInfo> newContracts = orgInfo.getOrgs().stream()
+                .filter(hpmsInfo -> !existingMap.containsKey(hpmsInfo.getContractId()))
+                .collect(Collectors.toList());
+        List<Contract> contractAttestList = addNewContracts(newContracts);
+
+        Map<String, HPMSOrganizationInfo> refreshed = buildRefreshedMap(orgInfo);
+        existingMap.forEach((contractId, contract) ->
+                considerContract(contractAttestList, contract, refreshed.get(contractId)));
+
+        batchAttestations(contractAttestList.stream().map(Contract::getContractNumber).collect(Collectors.toList()));
+    }
+
+    // Limit the size of the request to BATCH_SIZE, avoiding URLs that are too long and keeping the burden down
+    // on the invoked service.
+    private void batchAttestations(List<String> contractAttestList) {
+        final int size = contractAttestList.size();
+        int startIdx = 0;
+        for (; startIdx < size - BATCH_SIZE; startIdx += BATCH_SIZE) {
+            List<String> currentChunk = contractAttestList.subList(startIdx, startIdx + BATCH_SIZE);
+            processAttestations(currentChunk);
+        }
+
+        // process the remainder (if any) - i.e. smaller than a batch
+        if (size % BATCH_SIZE != 0) {
+            List<String> currentChunk = contractAttestList.subList(startIdx, size);
+            processAttestations(currentChunk);
+        }
+    }
+
+    private void processAttestations(List<String> currentChunk) {
+        String contractIdStr = currentChunk.stream().collect(Collectors.joining("\",\"", "[\"", "\"]"));
```

Comment:
You're using this string in joining in 2 places in the code, can you describe what the result in currentChunk you're expecting to get back will look like?

Reply:
Done.